### PR TITLE
Support for Python 3.11

### LINKED ISSUE #194: bug: Installation issues in Python 3.11, Linux Ubuntu 22.04
# Bug Report

**Fortuna version: **

0.1.45

**Current behavior:**

- On Ubuntu 22.04, after creating a new Conda environment with Python 3.11, and installation with `pip install aws-fortuna` (first install): Import of fortuna fails. This is apparently related to an issue with Flax and Jax versions.

**Expected behavior:**

Loading aws-fortuna

**Steps to reproduce:**

In a terminal, type the commands:
```
$ conda create --name test python=3.11
$ conda activate test
$ pip install aws-fortuna
$ python
>>> import fortuna
```

**Related code:**

```
Python 3.11.9 (main, Apr 19 2024, 16:48:06) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import fortuna
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/charles/miniconda3/envs/test/lib/python3.11/site-packages/fortuna/__init__.py", line 1, in <module>
    import flax
  File "/home/charles/miniconda3/envs/test/lib/python3.11/site-packages/flax/__init__.py", line 18, in <module>
    from .configurations import (
  File "/home/charles/miniconda3/envs/test/lib/python3.11/site-packages/flax/configurations.py", line 92, in <module>
    flax_filter_frames = define_bool_state(
                         ^^^^^^^^^^^^^^^^^^
  File "/home/charles/miniconda3/envs/test/lib/python3.11/site-packages/flax/configurations.py", line 42, in define_bool_state
    return jax_config.define_bool_state('flax_' + name, default, help)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'Config' object has no attribute 'define_bool_state'
>>> 
```

**Other information:**

On Linux, upgrading Jax and Flax to the latest versions actually solved the issue. I don't know if it will create further errors though...

### PR DESCRIPTION
**Pull request by wistuba**
Upgrades flax to 0.8.5 and updates code accordingly. Discontinues support for Python 3.8. Closes #194

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -35,6 +35,10 @@ torch = {version = "^2.1.0", optional = true}
 scikit-learn = {version = "^1.3.2", optional = true}
 umap-learn = {version = "^0.5.5", optional = true}
 tabulate = "^0.9.0"
+jax = "~0.4.30"
+jaxlib = "~0.4.30"
+tensorflow-io-gcs-filesystem = "0.29.0"
```
Comment (mdhanna) [inline pyproject.toml:40]: I was not able to install due to this requirement:
```
ERROR: Could not find a version that satisfies the requirement tensorflow-io-gcs-filesystem==0.29.0 (from aws-fortuna) (from versions: 0.34.0, 0.36.0, 0.37.0)
ERROR: No matching distribution found for tensorflow-io-gcs-filesystem==0.29.0
```

Comment (wistuba) [reply pyproject.toml:40]: can you share your OS and Python version?

Comment (mdhanna) [reply pyproject.toml:40]: MacOS and Python 3.11.9

Comment (wistuba) [reply pyproject.toml:40]: Thanks for pointing this out, I was able to reproduce this. This is happening with `pip`, right?

**Pull request merged**